### ISSUE TITLE
maven: Immutable classes do not exist after mvn generate-sources -> problem with querydsl

### ISSUE THREAD

**Issue by hubchenfox** [immutables/immutables#1385]
Steps to reproduce
1. Use Immutables (within JPA entities) and QueryDSL together
2. Run mvn clean generate-sources
3. Immutables are not generated in this step
4. QueryDSL (which generates the metamodel classes in the correct phase generate-sources) expects the Immutables and write error messages during the maven generate-sources phase

Note: The mvn install leads to a correct build, but error messages should be avoided

Wanted solution:
Immutable classes should be generated in the maven step "generate-sources"
This has also the advantage that even in case of compile errors the Immutables already exists and the "real error" does not hide between lots of "Immutable not found" errors.

**Comment by yatesco** [comment]
I think I ran into a similar problem https://github.com/typetools/checker-framework/issues/5143

**Comment by hubchenfox** [comment]
Quote @yatesco
> I think I ran into a similar problem [typetools/checker-framework#5143](https://github.com/typetools/checker-framework/issues/5143)
Thanks for the link.
As far as I understood it, it was about two different plugins both being executed in the maven lifecycle compile, but one of them was no longer executed.
Your solution from the link:
If you run "mvn clean generate-sources", the autogenerated classes of both plugins are "missing", aren't they?
I have now got QueryDSL to generate the classes in the compile lifecylce (annotationProcessor) only but this has other disadvantages and that is why I am not so happy with it

**Comment by yatesco** [comment]
Yes, it was exactly that. The solution was to use "auto discovery" of the annotation processors.
